src/PSDocs.Azure/docs/Azure.Template.Doc.ps1

```diff
@@ -20,15 +20,18 @@ function global:GetTemplateParameter {
                 Description = ''
                 DefaultValue = $Null
                 AllowedValues = $Null
+                Required = "Required"
             }
             if ([bool]$property.Value.PSObject.Properties['metadata'] -and [bool]$property.Value.metadata.PSObject.Properties['description']) {
                 $result.Description = $property.Value.metadata.description;
             }
             if ([bool]$property.Value.PSObject.Properties['defaultValue']) {
                 $result.DefaultValue = $property.Value.defaultValue;
+                $result.Required = "Optional"
             }
             if ([bool]$property.Value.PSObject.Properties['allowedValues']) {
                 $result.AllowedValues = $property.Value.allowedValues;
+                $result.Required = "Optional"
```

Comment:
@vicperdana Does setting `allowedValues` make a parameter optional?

Reply:
Fixed this in the latest update.